File: 3]. Competitive Programming/LeetCode/1]. Problems/Python/0035)_Search_Insert_Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)
        
        while left <= right and left <len(nums) and right >= 0:
            mid = (left+right)//2
            if target == nums[mid]:
                return mid
            if target < nums[mid]:
                right = mid-1
            else:
                left = mid+1
        return left

File: 3]. Competitive Programming/03]. LeetCode/1]. Problems/C++/0035)_Search_Insert_Position.cpp
class Solution 
{
  public:
    int searchInsert(vector<int>& nums, int target)
    {
        int low=0, high=nums.size();
        
        while(low<=high && low<nums.size() && high>=0)
        {
            int mid = (low+high)/2;
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] > target)
                high = mid-1;
            else
                low = mid+1;
        }
        return low;
    }
};
